Sources/Apodini/Semantic Model Builder/EndpointsTree.swift

```diff
@@ -37,11 +63,12 @@ struct Endpoint {
     let operation: Operation
 
     fileprivate var requestHandlerBuilder: (RequestInjectableDecoder) -> (Vapor.Request) -> EventLoopFuture<Vapor.Response>
+
     /// Type returned by `handle()`
-    let handleReturnType: ResponseEncodable.Type
+    var handleReturnType: ResponseEncodable.Type { HandleReturnType.self }
     /// Response type ultimately returned by `handle()` and possible following `ResponseTransformer`s
-    var responseType: ResponseEncodable.Type
-    
+    let responseType: ResponseEncodable.Type
+
     /// All `@Parameter` `RequestInjectable`s that are used inside handling `Component`
     var parameters: [EndpointParameter]
```

Comment:
is there a reason why this is mutable?

Reply:
Good catch. There is no reason for this to be mutable